homeassistant/components/system_bridge/coordinator.py

```diff
@@ -0,0 +1,137 @@
+"""DataUpdateCoordinator for System Bridge."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+import logging
+from typing import Callable
+
+from systembridge import Bridge
+from systembridge.client import BridgeClient
+from systembridge.exceptions import (
+    BridgeAuthenticationException,
+    BridgeConnectionClosedException,
+    BridgeException,
+)
+from systembridge.objects.events import Event
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_HOST,
+    CONF_PORT,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import BRIDGE_CONNECTION_ERRORS, DOMAIN
+
+
+class SystemBridgeDataUpdateCoordinator(DataUpdateCoordinator[Bridge]):
+    """Class to manage fetching System Bridge data from single endpoint."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        LOGGER: logging.Logger,
+        *,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize global System Bridge data updater."""
+        self.bridge = Bridge(
```

Comment:
We should probably only create a single client instance and not two. We've already created one in async_setup_entry.